### PR TITLE
Fix architectury 1.19.3

### LINKED ISSUE #875: 1.19.2 OptiFabric collision with JEI
Crash log:
```
[16:28:00] [main/INFO]: Loading Minecraft 1.19.2 with Fabric Loader 0.14.9
[16:28:00] [main/INFO]: Loading 86 mods:
	- appleskin 2.4.1+mc1.19
	- architectury 6.2.43
	- autofish 0.9.5-SNAPSHOT
	- badpackets 0.2.0
	- bookshelf 16.1.5
	- cloth-basic-math 0.6.1 via cloth-config
	- cloth-config 8.0.75
	- craftpresence 1.8.11
	- enchdesc 13.0.1
	- fabric-api 0.60.0+1.19.2
	- fabric-api-base 0.4.11+e62f51a390 via fabric-api
	- fabric-api-lookup-api-v1 1.6.9+9ff28f4090 via fabric-api
	- fabric-biome-api-v1 9.0.17+aeb40ebe90 via fabric-api
	- fabric-blockrenderlayer-v1 1.1.20+aeb40ebe90 via fabric-api
	- fabric-client-tags-api-v1 1.0.1+b35fea8390 via fabric-api
	- fabric-command-api-v1 1.2.11+f71b366f90 via fabric-api
	- fabric-command-api-v2 2.1.7+0c17ea9690 via fabric-api
	- fabric-commands-v0 0.2.28+df3654b390 via fabric-api
	- fabric-containers-v0 0.1.34+df3654b390 via fabric-api
	- fabric-content-registries-v0 3.2.3+aeb40ebe90 via fabric-api
	- fabric-convention-tags-v1 1.1.1+7cd20a1490 via fabric-api
	- fabric-crash-report-info-v1 0.2.5+aeb40ebe90 via fabric-api
	- fabric-data-generation-api-v1 5.1.8+a680b9b490 via fabric-api
	- fabric-dimensions-v1 2.1.30+aeb40ebe90 via fabric-api
	- fabric-entity-events-v1 1.4.18+9ff28f4090 via fabric-api
	- fabric-events-interaction-v0 0.4.28+aeb40ebe90 via fabric-api
	- fabric-events-lifecycle-v0 0.2.28+df3654b390 via fabric-api
	- fabric-game-rule-api-v1 1.0.21+aeb40ebe90 via fabric-api
	- fabric-item-api-v1 1.5.7+35a03c4390 via fabric-api
	- fabric-item-groups-v0 0.3.28+aeb40ebe90 via fabric-api
	- fabric-key-binding-api-v1 1.0.20+aeb40ebe90 via fabric-api
	- fabric-keybindings-v0 0.2.18+df3654b390 via fabric-api
	- fabric-language-kotlin 1.8.3+kotlin.1.7.10
	- fabric-lifecycle-events-v1 2.1.2+aeb40ebe90 via fabric-api
	- fabric-loot-api-v2 1.1.3+83a8659290 via fabric-api
	- fabric-loot-tables-v1 1.1.6+9e7660c690 via fabric-api
	- fabric-message-api-v1 5.0.3+176380a290 via fabric-api
	- fabric-mining-level-api-v1 2.1.14+33fbc73890 via fabric-api
	- fabric-models-v0 0.3.17+aeb40ebe90 via fabric-api
	- fabric-networking-api-v1 1.2.4+5eb68ef290 via fabric-api
	- fabric-networking-v0 0.3.21+df3654b390 via fabric-api
	- fabric-object-builder-api-v1 4.0.11+aeb40ebe90 via fabric-api
	- fabric-particles-v1 1.0.10+aeb40ebe90 via fabric-api
	- fabric-registry-sync-v0 0.9.24+aeb40ebe90 via fabric-api
	- fabric-renderer-api-v1 1.0.10+aeb40ebe90 via fabric-api
	- fabric-renderer-indigo 0.6.12+aeb40ebe90 via fabric-api
	- fabric-renderer-registries-v1 3.2.20+df3654b390 via fabric-api
	- fabric-rendering-data-attachment-v1 0.3.14+aeb40ebe90 via fabric-api
	- fabric-rendering-fluids-v1 3.0.7+aeb40ebe90 via fabric-api
	- fabric-rendering-v0 1.1.22+df3654b390 via fabric-api
	- fabric-rendering-v1 1.10.16+aeb40ebe90 via fabric-api
	- fabric-resource-conditions-api-v1 2.0.11+e62f51a390 via fabric-api
	- fabric-resource-loader-v0 0.6.1+aeb40ebe90 via craftpresence
	- fabric-screen-api-v1 1.0.26+f05effaf90 via fabric-api
	- fabric-screen-handler-api-v1 1.2.13+9ff28f4090 via fabric-api
	- fabric-textures-v0 1.0.20+aeb40ebe90 via fabric-api
	- fabric-transfer-api-v1 2.0.11+83a8659290 via fabric-api
	- fabric-transitive-access-wideners-v1 1.3.0+42d99c3290 via fabric-api
	- fabricloader 0.14.9
	- fiber 0.23.0-2 via neat
	- imblocker 1.0.19
	- inventoryhud 3.4.2
	- inventoryprofilesnext 1.7.1
	- java 17
	- jei 11.3.0.262
	- lightoverlay 7.0.0
	- minecraft 1.19.2
	- mm 2.3 via optifabric
	- modmenu 4.0.6
	- neat 1.19-32-FABRIC
	- optifabric 1.13.16
	- org_jetbrains_kotlin_kotlin-reflect 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk7 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk8 1.7.10 via fabric-language-kotlin
	- org_jetbrains_kotlinx_atomicfu-jvm 0.18.3 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.6.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.6.4 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-datetime-jvm 0.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.4.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.4.0 via fabric-language-kotlin
	- smoothboot 1.19-1.7.1
	- wthit 5.12.0
	- xaerominimap 22.14.0
	- xaeroworldmap 1.27.0
[16:28:03] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/F:/MultiMC/libraries/net/fabricmc/sponge-mixin/0.11.4+mixin.0.8.5/sponge-mixin-0.11.4+mixin.0.8.5.jar Service=Knot/Fabric Env=CLIENT
[16:28:03] [main/INFO]: Compatibility level set to JAVA_16
[16:28:03] [main/INFO]: Compatibility level set to JAVA_17
[16:28:04] [main/INFO]: Smooth Boot config initialized
[16:28:04] [main/WARN]: Configuration conflict: there is more than one oshi.properties file on the classpath
[16:28:04] [main/WARN]: Configuration conflict: there is more than one oshi.architecture.properties file on the classpath
[16:28:04] [main/WARN]: @Final field field_22786:Ljava/util/List; in mixins.ipnext.json:MixinScreen from mod inventoryprofilesnext should be final
[16:28:04] [main/WARN]: @Final field field_33815:Ljava/util/List; in mixins.ipnext.json:MixinScreen from mod inventoryprofilesnext should be final
[16:28:05] [main/ERROR]: Minecraft has crashed!
net.fabricmc.loader.impl.FormattedException: java.lang.ExceptionInInitializerError
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:463) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.9.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
Caused by: java.lang.ExceptionInInitializerError
	at net.minecraft.class_128.method_568(class_128.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_128.method_24305(class_128.java:345) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:159) ~[minecraft-1.19.2-client.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-1.19.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.9.jar:?]
	... 10 more
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_309 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:419) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.9.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.Class.getDeclaredFields0(Native Method) ~[?:?]
	at java.lang.Class.privateGetDeclaredFields(Class.java:3297) ~[?:?]
	at java.lang.Class.getDeclaredFields(Class.java:2371) ~[?:?]
	at net.optifine.reflect.FieldLocatorTypes.<init>(FieldLocatorTypes.java:25) ~[Optifine-mapped.jar:?]
	at net.optifine.reflect.Reflector.<clinit>(Reflector.java:501) ~[Optifine-mapped.jar:?]
	at net.minecraft.class_128.method_568(class_128.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_128.method_24305(class_128.java:345) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:159) ~[minecraft-1.19.2-client.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-1.19.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.9.jar:?]
	... 10 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:414) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.9.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.Class.getDeclaredFields0(Native Method) ~[?:?]
	at java.lang.Class.privateGetDeclaredFields(Class.java:3297) ~[?:?]
	at java.lang.Class.getDeclaredFields(Class.java:2371) ~[?:?]
	at net.optifine.reflect.FieldLocatorTypes.<init>(FieldLocatorTypes.java:25) ~[Optifine-mapped.jar:?]
	at net.optifine.reflect.Reflector.<clinit>(Reflector.java:501) ~[Optifine-mapped.jar:?]
	at net.minecraft.class_128.method_568(class_128.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_128.method_24305(class_128.java:345) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:159) ~[minecraft-1.19.2-client.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-1.19.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.9.jar:?]
	... 10 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Callback method beforeCharTypedEvent(Lnet/minecraft/class_364;IILorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;)V in jei.mixins.json:KeyboardHandlerMixin from mod jei failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap jei-1.19.2-fabric-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1385) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.11.4+mixin.0.8.5.jar:0.11.4+mixin.0.8.5]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:414) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145) ~[fabric-loader-0.14.9.jar:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at java.lang.Class.getDeclaredFields0(Native Method) ~[?:?]
	at java.lang.Class.privateGetDeclaredFields(Class.java:3297) ~[?:?]
	at java.lang.Class.getDeclaredFields(Class.java:2371) ~[?:?]
	at net.optifine.reflect.FieldLocatorTypes.<init>(FieldLocatorTypes.java:25) ~[Optifine-mapped.jar:?]
	at net.optifine.reflect.Reflector.<clinit>(Reflector.java:501) ~[Optifine-mapped.jar:?]
	at net.minecraft.class_128.method_568(class_128.java:175) ~[client-intermediary.jar:?]
	at net.minecraft.class_128.method_24305(class_128.java:345) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.method_44604(Main.java:159) ~[minecraft-1.19.2-client.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[minecraft-1.19.2-client.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) ~[fabric-loader-0.14.9.jar:?]
	... 10 more

```

### LINKED ISSUE #978: Incompatibility with architectury mod "Roughly Enough Items" dependency OR "Just Enough Items" with OptiFabric 1.19.3 beta
OptiFabric 1.19.3 version is not compatible with two mods, Here is the comments I sent on "1.19.3 Support" issue, I thought it was the best thing to create a separate issue.

I don't understand much but the problem seems to be incompatibility with architectury.

Well the log is the same in file or text, But if you care about it that much here it is: https://github.com/Chocohead/OptiFabric/files/10832471/latest.log

I tested that theory, and it was right, OptiFabric 1.19.3 beta is not compatible with architectury mod, And that is required for

"Roughly Enough Items". Without "Roughly Enough Items" and it's dependencies like "architectury", The game runs fine.
OptiFabric 1.19.3 is not compatible with "Roughly Enough Items" or
"Just Enough Items" here is the log for another crash related to "Just Enough Items" and OptiFabric, Tested without Optifabric and the game loads fine, tested with optifabric and
without "Just Enough Items" and the game loads fine, The issue is "Just Enough Items" with OptiFabric.
Just enough items log: https://mclo.gs/jBSOHDk

### LINKED ISSUE #996: 1.19.3 optifabric missing "mm 2.0"
I have no idea what this mod is or where to find it. I've searched and can't find any mention of it.

I am using 1.13.21-dev, 1.13.21-sources-dev and 1.19.3_HD_U_I2_pre5.

I have also tried using all the other variants of Optifabric. 1.13.21 just crashes on launch. I have tried without the sources and got the same result.

### PR DESCRIPTION
**Pull request by XXMA16**
### Regarding the fix
1.19.3 switched to `org.joml.Matrix4f` and the target of `renderScreenPre` and `renderScreenPost` was also changed. Due to the fact that these changes were made in different Architectury API versions, anything strictly below `7.0.59` **will crash** _even though the version range for the new patches is `>=7.0.52`_. While not particularly hard to do so, these crashes should not be fixed.
As far as I checked, Architectury should run just fine on 1.19.2 and the latest 1.19.3 version.

### Regarding the project structure
I think the naming scheme **needs** an overhaul as keeping track of all the files named _Newererer_ is quite difficult. On another note, there _might_ be a way of automatically generating the required Intercepting Mixins at build time, without explicitly creating each one of them, thus requiring less maintenance for compat patches. Furthermore, a fake `LanguageAdapter` class could perform some kind of analysis and intercept mixins based on it.

fixed #978 
fixed #875
fixed #996
_Might be some others but I'm too lazy to check_

### CODE REVIEW

**Comment by topia** [comment]
For curious, if it crashed with below 7.0.59, why should we check `>=7.0.52` instead of `>=7.0.59`?

**Comment by XXMA16** [comment]
> For curious, if it crashed with below 7.0.59, why should we check `>=7.0.52` instead of `>=7.0.59`?

It would have crashed anyway because this is the version that made the switch to `org.joml.Matrix4f`. The other change (from `render` to `renderWithTooltip`) took place at a later time in an (as far as I know) unreleased version.

**Comment by XXMA16** [comment]
**Might as well fix JEI too**
I have tested 1.19.3 and 1.19.2 and both of the seem to work just fine.
_The Optifine lambdas take a `Screen` as the first parameter, while the Vanilla ones take an `Element`, which is an interface implemented by `Screen`. As far as I tested this is not an issue, but I am unsure whether or not it will prove to be one in the future._

**Comment by Dracconis** [comment]
So when will this patch be available as optifabric still crashes with the 1.13.21 version using the newerer instead of the aforementioned newererer (yes 3 -er's)

**Comment by XXMA16** [comment]
> So when will this patch be available as optifabric still crashes with the 1.13.21 version using the newerer instead of the aforementioned newererer (yes 3 -er's)

You can get the artifacts (jar files) by going to [Actions](https://github.com/Chocohead/OptiFabric/actions), clicking on the commit/pull request that made the changes and scrolling down to the _Artifacts_ section. Download the zip and extract the jar that does **not** have `dev` nor `sources` in its name. For this specific pull request you may find the artifacts [here](https://github.com/Chocohead/OptiFabric/suites/11660493196/artifacts/605777007).

**Comment by Dracconis** [comment]
Thank you sir for the information and everything works great now.

**Comment by GamingGuyRob** [comment]
I dont want to open another issue for this, but I have the latest 1.13.21 from the actions. Im also using 1.19.3_pre5 and I still crash on launch.

The log file:

https://paste.ee/p/NcxHD

p.s. - maybe you should hire me to test because I'm clearly good at breaking things haha

**Comment by Dracconis** [comment]
You are still using the old version of Optifabric 1.13.21 as it states newerer instead of newererer in the error.
You need the one from here: https://github.com/Chocohead/OptiFabric/actions/runs/4460930918

**Comment by GamingGuyRob** [comment]
> You are still using the old version of Optifabric 1.13.21 as it states newerer instead of newererer in the error. You need the one from here: https://github.com/Chocohead/OptiFabric/actions/runs/4460930918

Still crashing using the file you linked

https://paste.ee/p/Tc4FA

Am I supposed to be using both optifabric and the sources?

**Comment by Dracconis** [comment]
Ok now you are getting a different crash error. I would suggest removing the "optifabric-1.13.21-sources.jar" from your mods and try again as that mod is not needed. Also IF it still crashes then only have the Fabric API, Optifabric, Optifine, Architectury, and only 1 of the mods that require Architectury and see if it will run. IF it does then add 1 mod back in and try again. Keep doing that until it crashes and you will then know which mod is causing the conflict.

**Comment by XXMA16** [comment]
> > You are still using the old version of Optifabric 1.13.21 as it states newerer instead of newererer in the error. You need the one from here: https://github.com/Chocohead/OptiFabric/actions/runs/4460930918
> 
> Still crashing using the file you linked
> 
> https://paste.ee/p/Tc4FA
> 
> Am I supposed to be using both optifabric and the sources?

TLDR: Use Fabric API 0.75.1 until an official fix for Architectury is released by the repository owner.

The crash is caused by Fabric API 0.76.0. This was fixed in the latest commit made by @Chocohead, however it does not include the Architectuy fix since the pull request hasn't been merged. The again, my approach might be dumb and cause even more issues in the future or Chocohead might not have had the time necessary to review my PR.

**Comment by Chocohead** [comment]
> The again, my approach might be dumb and cause even more issues in the future or Chocohead might not have had the time necessary to review my PR.

For Architectury it's exactly the fix it needs. For future reference the change to `Matrix4f` came in architectury/architectury-api@c3f3d962 whilst the target change happened in architectury/architectury-api/commit/3e5fdd4b. Given that there's no OptiFine version for 22w43a nor 1.19.3-pre1 it's safe to just ignore the gap in-between changes.

For JEI it's a bit more subjective. OptiFine manages to mess up pretty much all the `Keyboard` lambdas by including Forge's changes (and generally just recompiling reordering parameters), but it does make its own changes too so skipping patching `Keyboard` completely isn't a good option. #1005 is caused by `KeyboardFix` not fixing these problems well enough, thus care is needed to get things right.

> I think the naming scheme **needs** an overhaul as keeping track of all the files named _Newererer_ is quite difficult.

In fairness it was #630 which introduced `newerer` :P It's quite funny when you're not trying to find the file with the right number of `er`s. The Mixin files could do with grouping into directories anyway. This PR brings the total to 90 which is almost as much to blame for being unwieldly as the individual file names.

> On another note, there _might_ be a way of automatically generating the required Intercepting Mixins at build time, without explicitly creating each one of them, thus requiring less maintenance for compat patches.

I could believe it. Especially with the `Matrix4f` ones, all that's changed is one import. I had been looking at more generic fixes which mod compat could just choose to include. `GameRenderer` and `WorldRenderer` both have multiple very similar copies of the same fix, so being able to only have one instance of each fix would certainly improve maintainability compared to the current situation. It could also make fixing other mods faster (were they to also need the same kind of fix).

> Furthermore, a fake `LanguageAdapter` class could perform some kind of analysis and intercept mixins based on it.

I think you'd end up spending more time each launch trying to detect where fixes were needed than you'd save not having to manually make them. You'd have to explore each mod's jar (every launch) and evaluate whether what each mixin was doing was going to be a problem which would be slow and error prone. I think the better avenue is getting fixes to be more generic, thus easier for new mod compat to use, and a more robust testing system for making sure mods will work as they and OptiFine/Fabric update.
